src/main/java/com/zufar/onlinestore/payment/service/PaymentEventHandler.java

```diff
@@ -0,0 +1,44 @@
+package com.zufar.onlinestore.payment.service;
+
+import com.stripe.model.Event;
+import com.stripe.model.PaymentIntent;
+import com.zufar.onlinestore.payment.entity.Payment;
+import com.zufar.onlinestore.payment.enums.PaymentStatus;
+import com.zufar.onlinestore.payment.repository.PaymentRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import java.util.Objects;
+
+@Slf4j
+@RequiredArgsConstructor
+@Service
+public class PaymentEventHandler {
+
+    private final PaymentRepository paymentRepository;
+
+    public void handleEvent(Event event, PaymentIntent paymentIntent) {
+        if (!Objects.isNull(paymentIntent) && !Objects.isNull(event)) {
+            String paymentIntentId = paymentIntent.getId();
+            String eventType = event.getType();
+            log.info("handle event: preparation for payment event status handling: " +
+                    "paymentIntentId: {}, eventType: {}.", paymentIntentId, eventType);
+
+            if (Objects.equals(PaymentStatus.PAYMENT_IS_SUCCEEDED.getStatus(), eventType)) {
```

Comment:
I think it would be better if you change it this way:

```
   PaymentStatus status = null;
   switch (eventType) {
   
     case -> PaymentStatus.PAYMENT_IS_SUCCEEDED {
      status = PaymentStatus.PAYMENT_IS_SUCCEEDED;
      break;
     };
    case -> PaymentStatus.PAYMENT_IS_FAILED {
      status = PaymentStatus.PAYMENT_IS_FAILED;
      break;
     };
     default {
      throw new UnknownEventTypeException("EventType = {} is unknown", eventType);
     }
   
   }
```

Reply:
Done